v2/cmd/asoctl/cmd/import_azure_resource.go

```diff
@@ -75,26 +83,62 @@ func importAzureResource(ctx context.Context, armIDs []string, outputPath *strin
 	result, err := importer.Import(ctx)
 
 	// Wait for progress bar to finish & flush
-	progress.Wait()
+	defer func() {
+		progress.Wait()
+	}()
 
 	if err != nil {
 		return errors.Wrap(err, "failed to import resources")
 	}
 
-	if outputPath == nil || *outputPath == "" {
-		err := result.SaveToWriter(os.Stdout)
+	if result.Count() == 0 {
+		log.Info("No resources found, nothing to save.")
+		return nil
+	}
+
+	if file, ok := options.writeToFile(); ok {
+		log.Info(
+			"Writing to a single file",
+			"file", file)
+		err := result.SaveAllToFile(file)
 		if err != nil {
-			return errors.Wrapf(err, "failed to write to stdout")
+			return errors.Wrapf(err, "failed to write to file %s", file)
 		}
-	} else {
+	} else if folder, ok := options.writeToFolder(); ok {
 		log.Info(
-			"Writing to file",
-			"path", *outputPath)
-		err := result.SaveToFile(*outputPath)
+			"Writing to individual files in folder",
+			"folder", folder)
+		err := result.SaveToFolder(folder)
```

Comment:
minor: pedant in me but one of these save methods is `SaveAllToFile` and one is `SaveToFolder`. Are we not saving `All` to folder?

Feels like they should mirror one another and either both have `All` or neither have `All`.

Reply:
I was trying to reflect the differing semantics in the names - one creates a single file, the other creates _n_ files. I'll rename the methods.

Reply:
Renamed to `SaveToSingleFile()` and `SaveToIndividualFilesInFolder()` which are a bit unwieldy, but clear.